sycl/source/detail/thread_pool.hpp

```diff
@@ -91,7 +102,7 @@ class ThreadPool {
       std::lock_guard<std::mutex> Lock(MJobQueueMutex);
       MJobQueue.emplace(Func);
     }
-
+    MJobsInExecution++;
```

Comment:
Shouldn't the counter be incremented in the another version of `submit` as well.

Reply:
you are right, missed that we have two versions and got hang(
fixed in https://github.com/intel/llvm/pull/6837/commits/06e2608dfccb0c349a1e6aa25fad727c69ed7155